src/Sulu/Bundle/ContentBundle/Resources/config/content_types.xml

```diff
@@ -20,6 +20,7 @@
             <argument type="service" id="sulu_content.smart_content.data_provider.content.query_builder"/>
             <argument type="service" id="logger"/>
             <argument>%sulu.content.type.internal_links.template%</argument>
+            <argument type="expression">parameter('sulu.context') === 'website' and (!container.hasParameter('sulu.preview') or !parameter('sulu.preview'))</argument>
```

Comment:
Not sure if there is a better way to do this... I'd like to not simply pass the context and preview to the content type, because then it is coupled to these values. This way the component has only a flag describing if it should show the non-published pages or not. IMO this is cleaner.

However, I don't know where else I should put this logic, although I don't really want to have to complicated service definitions... On the other side the parameters are also configured in the service container, so it seems to be a good choice. 🤔